services/QuillLMS/app/workers/onboarding_checklist_analytics_worker.rb

```diff
@@ -0,0 +1,14 @@
+class OnboardingChecklistAnalyticsWorker
+  include Sidekiq::Worker
+
+  def perform(user_id)
+    all_objectives_met = Objective::ONBOARDING_CHECKLIST_NAMES.all? do |name|
+      objective = Objective.find_by_name(name)
+      Checkbox.find_by(user_id: user_id, objective_id: objective.id)
```

Comment:
This is a super-minor note that's just a call-out of Rails syntax.  You can make these two lines into one:
```ruby
Checkbox.joins(:objective).find_by(user_id: user_id, objectives: {name: name})
```
It's technically less load on the db (though to a trivial degree here), so I figured I'd point it out as an option, but it's also a little trickier to read/understand which means I'm not actually pushing for a change here.